thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/common/BaseFactory.java

```diff
@@ -0,0 +1,13 @@
+package com.linkedin.thirdeye.common;
+
+public class BaseFactory<T> {
```

Comment:
1. Please format the code.
2. Should BaseFactory considering package name when creating an instance? What about making base factory creates instances from any full class names. The factory that extends the base factory could override the method to inject package name if necessary.